Sources/SuperwallKit/Dependencies/DependencyContainer.swift

```diff
@@ -0,0 +1,344 @@
+//
+//  File.swift
+//  
+//
+//  Created by Yusuf Tör on 23/12/2022.
+//
+// swiftlint:disable function_body_length
+
+import UIKit
+
+/// Contains all of the SDK's core utility objects that are normally directly injected as dependencies.
+///
+/// This conforms to protocol factory methods which can be used to make objects that have
+/// dependencies injected into them.
+///
+/// Objects only need `unowned` references to the dependencies injected into them because
+/// `DependencyContainer` is owned by the `Superwall` class.
+///
+/// Idea taken from: [swiftbysundell.com](https://www.swiftbysundell.com/articles/dependency-injection-using-factories-in-swift/)
+final class DependencyContainer {
+  // swiftlint:disable implicitly_unwrapped_optional
+  var configManager: ConfigManager!
+  var identityManager: IdentityManager!
+  var storeKitManager: StoreKitManager!
+  var appSessionManager: AppSessionManager!
+  var sessionEventsManager: SessionEventsManager!
+  var storage: Storage!
+  var network: Network!
+  var paywallManager: PaywallManager!
+  var paywallRequestManager: PaywallRequestManager!
+  var deviceHelper: DeviceHelper!
+  var localizationManager: LocalizationManager!
+  var queue: EventsQueue!
+  var debugManager: DebugManager!
+  var api: Api!
+  var transactionManager: TransactionManager!
+  var restorationHandler: RestorationHandler!
+  var delegateAdapter: SuperwallDelegateAdapter!
+  // swiftlint:enable implicitly_unwrapped_optional
+
+  init(
+    apiKey: String,
+    swiftDelegate: SuperwallDelegate? = nil,
+    objcDelegate: SuperwallDelegateObjc? = nil,
+    options: SuperwallOptions? = nil
+  ) {
+    storeKitManager = StoreKitManager(factory: self)
+    delegateAdapter = SuperwallDelegateAdapter(
+      swiftDelegate: swiftDelegate,
+      objcDelegate: objcDelegate
+    )
+    localizationManager = LocalizationManager()
+    storage = Storage()
+    network = Network(factory: self)
+
+    paywallRequestManager = PaywallRequestManager(
+      storeKitManager: storeKitManager
+    )
+    paywallManager = PaywallManager(
+      factory: self,
+      paywallRequestManager: paywallRequestManager
+    )
+
+    configManager = ConfigManager(
+      storeKitManager: storeKitManager,
+      storage: storage,
+      network: network,
+      paywallManager: paywallManager,
+      factory: self
+    )
+
+    if let options = options {
+      configManager.options = options
+    }
+
+    api = Api(configManager: configManager)
```

Comment:
This brings me on to setting the options after/before config. People can do `Superwall.options.property = something` before they call `Superwall.configure`. It's a convenience thing that Jake liked. They can technically do it after too. What are your thoughts on that? This is why I have to pass in the configmanager options, rather than the options directly. But yeah moved away from passing in the actual configManager now.